ngraph/frontend/paddlepaddle/src/op/pool2d.cpp

```diff
@@ -25,47 +15,227 @@ namespace ngraph
+                                {std::make_shared<ngraph::opset6::MaxPool>(
+                                    data,
+                                    ngraph::Strides({1, 1}),
+                                    ngraph::Shape{0, 0}, // FIXME pads_begin
+                                    ngraph::Shape{0, 0}, // pads_end
+                                    ngraph::Shape{input_h, input_w})},
+                                {"Out"});
+                        }
+                        else
+                        {
+                            auto axes = ngraph::opset6::Constant::create(
+                                ngraph::element::i64, {2}, {input_rank - 2, input_rank - 1});
+                            return node.default_single_output_mapping(
+                                {std::make_shared<ngraph::opset6::ReduceMean>(data, axes, true)},
+                                {"Out"});
+                        }
                     }
-                    else if (pooling_type == "avg" &&
-                             (global_pooling ||
-                              (adaptive && all_of(kernel_shape.begin(),
-                                                  kernel_shape.end(),
-                                                  [](int32_t s) { return s == 1; }))))
+                    else if (adaptive)
                     {
-                        // TODO : resolve axes according to rank
-                        auto axes =
-                            ngraph::opset6::Constant::create(ngraph::element::i64, {2}, {2, 3});
-                        return node.default_single_output_mapping(
-                            {std::make_shared<ngraph::opset6::ReduceMean>(data, axes, true)},
-                            {"Out"});
+                        uint64_t pool_size_Height, pool_size_Width;
+                        if (kernel_shape.size() == 1)
+                        {
+                            pool_size_Height = pool_size_Width = kernel_shape[0];
+                        }
+                        else
+                        {
+                            pool_size_Height = kernel_shape[0];
+                            pool_size_Width = kernel_shape[1];
+                        }
+
+                        uint64_t stride_h = int64_t(input_h / pool_size_Height);
+                        uint64_t stride_w = int64_t(input_w / pool_size_Width);
+                        uint64_t kernel_h = input_h - (pool_size_Height - 1) * stride_h;
+                        uint64_t kernel_w = input_w - (pool_size_Width - 1) * stride_w;
+
+                        if (stride_h < 1 || stride_w < 1)
+                        { // upsampling?
+                            throw std::runtime_error("Unsupported pooling adaptive type!");
+                        }
+
+                        if (pooling_type == "max")
+                        {
+                            std::cout << "Convert MAx" << std::endl;
```

Comment:
avoid std::cout

Reply:
@ceciliapeng2011 please remove this cout